homeassistant/components/fibaro/__init__.py

```diff
@@ -178,11 +182,31 @@ def _on_state_change(self, state: Any) -> None:
             for callback in self._callbacks[item]:
                 callback()
 
+        resolver = FibaroStateResolver(state)
+        for event in resolver.get_events():
+            fibaro_id = event.fibaro_id
+            if (
+                event.event_type.lower() == "centralsceneevent"
+                and fibaro_id in self._event_callbacks
+            ):
+                for callback in self._event_callbacks[fibaro_id]:
+                    callback(event)
+
     def register(self, device_id: int, callback: Any) -> None:
         """Register device with a callback for updates."""
         self._callbacks.setdefault(device_id, [])
         self._callbacks[device_id].append(callback)
 
+    def register_event(
+        self, device_id: int, callback: Callable[[FibaroEvent], None]
+    ) -> None:
+        """Register device with a callback for central scene events.
+
+        The callback receives one parameter with the event.
+        """
+        self._event_callbacks.setdefault(device_id, [])
```

Comment:
`dict.setdefault` returns the value of the key. We can store the value in a local variable so we can use it on the next line and don't need to access the attribute twice.